File: Ruby/Common_Sorting_Algorithms/insertionSort/insertionsort.rb
def insertionSort(arr)
  i = 0
  while i < arr.length
    keepGoing = true
    j = i - 1
    while j >= 0 && keepGoing
      item = arr[j + 1]
      if item >= arr[j]
        keepGoing = false
      end
      if item < arr[j]
        arr[j + 1] = arr[j]
        arr[j] = item
      end
      j -= 1
    end

    i += 1
  end
  arr
end


p insertionSort([5,1,9,6,3,45,7])

File: JavaScript/Common_Algorithms/insertionSort/insertionSort.js
function insertionSort(arr){
	for(var i = 0; i < arr.length; i++){
		var keepGoing = true
		for(var j = i - 1; j>= 0 && keepGoing; j--){
			let item = arr[j + 1]
			if (item >= arr[j]){
				keepGoing = false
			}
			if (item < arr[j]){
				arr[j + 1] = arr[j]
				arr[j] = item
			}
		}
	}
	return arr
}

console.log(insertionSort([24,123,4,1,6,3,91]))
